src/modules/dashboard/lists/SlippageList.tsx

```diff
@@ -0,0 +1,134 @@
+import { CheckIcon, ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/outline';
+import { Trans } from '@lingui/macro';
+import {
+  Box,
+  Button,
+  Divider,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+  SvgIcon,
+  Typography,
+} from '@mui/material';
+import * as React from 'react';
+import { SlippageTooltip } from 'src/components/infoTooltips/SlippageTooltip';
+
+import Paraswap from '/public/icons/other/paraswap.svg';
+
+interface ListSlippageButtonProps {
+  setSlippage: (value: string) => void;
+  selectedSlippage: string;
+}
+
+export const ListSlippageButton = ({ setSlippage, selectedSlippage }: ListSlippageButtonProps) => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const values: string[] = ['0.1', '0.5', '1'];
+
+  return (
+    <>
+      <Button
+        variant="text"
+        onClick={handleClick}
+        size="medium"
+        endIcon={
+          <SlippageTooltip
+            text={
+              <Box sx={{ display: 'flex', alignItems: 'center' }}>
+                <Trans>
+                  <Typography color="text.secondary" variant="description">
+                    Slippage tolerance{' '}
+                  </Typography>
+                  <Typography color="text.main" variant="secondary14" sx={{ px: '4px' }}>
+                    {selectedSlippage}%{' '}
+                  </Typography>
+                  <SvgIcon sx={{ fontSize: '14px !important', mr: '4px' }}>
+                    {open ? <ChevronUpIcon /> : <ChevronDownIcon />}
+                  </SvgIcon>
+                </Trans>
+              </Box>
+            }
+            variant="secondary14"
+          />
+        }
+        disabled={false}
+        data-cy={`slippageButton_${selectedSlippage}`}
+        sx={{ mt: 6 }}
+      />
+
+      <Menu
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        MenuListProps={{
+          'aria-labelledby': 'basic-button',
+        }}
+        keepMounted={true}
+        data-cy={`slippageMenu_${selectedSlippage}`}
+      >
+        <Box sx={{ px: '16px', py: '12px' }}>
+          <Typography variant="secondary12" color="text.secondary">
+            <Trans>Select slippage tolerance</Trans>
+          </Typography>
+        </Box>
+
+        {values.map((slippageValue) => {
+          const selected = slippageValue === selectedSlippage;
```

Comment:
Is a separate const necessary here, would `selected={sliappageValue === selectedSlippage}` suffice?

Reply:
I think it was pulled out because the same check is done a few lines below